src/app/pages/rss/rss-feeds/rss-feeds.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, Input } from '@angular/core';
+import { RssFeedService } from '../../../services/rss.service';
+
+@Component({
+  selector: 'app-rss-feeds',
+  templateUrl: './rss-feeds.component.html',
+  styleUrls: ['./rss-feeds.component.scss'],
+})
+export class RssFeedsComponent {
+  @Input() rssLink!: string;
+  rssData?: any;
+  activeCardIndex = 0;
+
+  constructor(private rssFeedService: RssFeedService) {}
+
+  ngOnInit() {
+    this.getRssData();
+  }
+  getRssData() {
+    this.rssFeedService.getRssData(this.rssLink).subscribe({
+      next: (response: object) => {
+        this.rssData = response;
+        console.log(this.rssData);
+      },
+      error: (error: any) => {
+        console.error('Erreur lors de la récupération du flux RSS', error);
+      },
+    });
+  }
+
+  getTruncated(description: string, maxLength: number): string {
```

Comment:
can be easily done with css to avoid unnecessary js see [doc here](https://css-tricks.com/snippets/css/truncate-string-with-ellipsis/)